content/blog/2019-08-alphas.md

```diff
@@ -0,0 +1,232 @@
+- [Facebook](https://github.com/facebookexperimental/mononoke)
+- [Vector](https://vector.dev/)
+- [Deno](https://github.com/denoland/deno)
+- [Libra](https://github.com/libra/libra)
+- [PingCAP](https://pingcap.com)
+- [Parity](https://www.parity.io/)
+
+Over the past three years, the Rust team has not been idle either. They have
+been working tirelessly on a huge feature that will greatly improve the
+experience of writing asynchronous code with Rust: [async & await
+syntax][async_await]. This new syntax is about to land on Rust stable.
+
+All of Tokio's iteration to date was done while maintaining API stability. Tokio
+has not had a breaking release since the very first 0.1.0 release on September
+9, 2016. Understandibly, a long list of changes to make has been building up.
+The introduction of async & await syntax is the right moment to make these
+breaking changes.
+
+# What we have now
+
+Today's release of Tokio 0.2.0-alpha.1 with async & await updates all of Tokio's
+components to use [`std::future`][std_future] and all usage of `futures` 0.1 has
+been removed.
+
+Changes include:
+
+**Using async fns for asynchronous operations.**
+
+When possible, `async fn` has been used to provide asynchronous operations. For
+example, acceping sockets from a `TcpListener` is done with the async `accept`
+function:
+
+```rust,ignore
+let (mut socket, _) = listener.accept().await?;
+```
+
+I/O operations are provided as `async` functions:
+
+```rust,ignore
+let n = socket.read(&mut buf).await?;
+
+let n = socket.write(&buf).await?;
+```
+
+The same applies to most of the Tokio 0.2 APIs.
+
+**Starting the Tokio runtime**
+
+Setting the entry point of a Tokio application is now done with a proc macro:
+
+```rust,ignore
+#[tokio::main]
+async fn main() {
+    println!("Hello from Tokio!");
+}
+```
+
+Creating a runtime by hand is still supported. The proc macro just provides some
+sugar. The equivalent application would be:
+
+```rust,ignore
+fn main() {
+    let mut rt = Runtime::new().unwrap();
+    rt.block_on(async {
+        println!("Hello from Tokio!");
+    });
+}
+```
+
+There is also a proc macro for tests:
+
+```rust,ignore
+#[tokio::test]
+async fn my_test() {
+    let addr = "127.0.0.1:8080".parse().unwrap();
+    let mut listener = TcpListener::bind(&addr).unwrap();
+    let addr = listener.local_addr().unwrap();
+
+    // Connect to the listener
+    TcpStream::connect(&addr).await.unwrap();
+}
+```
+
+**Nightly only**
+
+Because async & await are not available on the stable Rust channel yet, this
+alpha release requires using nightly Rust. Tokio includes a
+[rust-toolchain][rustup] file that tracks the nightly Tokio tests again.
+
+[rustup]: https://github.com/tokio-rs/tokio/blob/master/rust-toolchain
+
+# Towards a final release
+
+This alpha release represents the first step towards the next iteration of
+Tokio. Users are now able to experiment using Tokio and async / await syntax,
+but nothing is set in stone.
```

Comment:
This might be better to say?

```suggestion
but traits like `Stream` and `Sink` have yet to be stabilized.
```

Reply:
I'm mostly trying to say that everything is in flux / nothing is final.

Reply:
well I think async/await is set in stone, but the rest isnt since this is a mvp? Im fine either way just wanted to maybe make this clear what this means.